nexus_constructor/InstrumentView.py

```diff
@@ -0,0 +1,157 @@
+from PySide2.QtWidgets import QWidget, QVBoxLayout
+from PySide2.Qt3DExtras import Qt3DExtras
+from PySide2.Qt3DCore import Qt3DCore
+from PySide2.QtCore import QPropertyAnimation
+from PySide2.QtGui import QVector3D, QColor, QMatrix4x4
+from nexus_constructor.NeutronAnimationController import NeutronAnimationController
+
+
+class InstrumentView(QWidget):
+    def __init__(self, parent):
+        super().__init__()
+        lay = QVBoxLayout(self)
+        self.view = Qt3DExtras.Qt3DWindow()
+        self.view.defaultFrameGraph().setClearColor(QColor("lightgrey"))
+        container = QWidget.createWindowContainer(self.view)
+        lay.addWidget(container)
+
+        self.view.camera().lens().setPerspectiveProjection(45, 16 / 9, 0.1, 1000)
+        self.view.camera().setPosition(QVector3D(6, 8, 30))
+        self.view.camera().setViewCenter(QVector3D(0, 0, 0))
+
+        self.rootEntity = Qt3DCore.QEntity()
+        camera_entity = self.view.camera()
+        cam_controller = Qt3DExtras.QFirstPersonCameraController(self.rootEntity)
+        cam_controller.setLinearSpeed(20)
+        cam_controller.setCamera(camera_entity)
+        self.view.setRootEntity(self.rootEntity)
+
+        # Initialise materials
+        self.grey_material = Qt3DExtras.QPhongMaterial()
+        self.red_material = Qt3DExtras.QPhongMaterial()
+        self.beam_material = Qt3DExtras.QPhongAlphaMaterial()
+        self.green_material = Qt3DExtras.QPhongMaterial()
+
+        # Initialise cube objects
+        self.cube_entity = Qt3DCore.QEntity(self.rootEntity)
+        self.cube_mesh = Qt3DExtras.QCuboidMesh()
+
+        # Create lists for neutron-related objects
+        self.neutron_entities = []
+        self.neutron_meshes = []
+        self.neutron_transforms = []
+        self.neutron_animation_controllers = []
+        self.neutron_animations = []
+
+        self.num_neutrons = 9
+        self.cylinder_length = 40
+
+        self.initialise_view()
+
+        # Initialise beam objects
+        self.cylinder_entity = Qt3DCore.QEntity(self.rootEntity)
+        self.cylinder_mesh = Qt3DExtras.QCylinderMesh()
+        self.cylinder_transform = Qt3DCore.QTransform()
+
+        self.create_beam_cylinder()
+
+    def create_materials(self):
+        """
+        Creates the materials for the the objects that will inhabit the instrument view.
+        """
+        red = QColor("red")
+        black = QColor("black")
+        grey = QColor("grey")
+        blue = QColor("blue")
+        light_blue = QColor("lightblue")
+        dark_red = QColor("#b00")
+
+        self.grey_material.setAmbient(black)
```

Comment:
looks like the following lines are almost duplication, could refactor this out into a separate function and test that it applies the ambient and diffuse values.